cmd/check.go

```diff
@@ -0,0 +1,318 @@
+package cmd
+
+import (
+	"context"
+	"encoding/xml"
+	"errors"
+	"fmt"
+	"io/fs"
+	"log"
+	"os"
+	"path"
+	"path/filepath"
+	"strconv"
+	"strings"
+
+	"github.com/golang-migrate/migrate/v4"
+	// needed driver.
+	_ "github.com/golang-migrate/migrate/v4/database/postgres"
+	_ "github.com/golang-migrate/migrate/v4/source/file"
+	"github.com/spf13/cobra"
+
+	"github.com/stack11/trek/internal"
+	"github.com/stack11/trek/internal/dbm"
+)
+
+func NewCheckCommand() *cobra.Command {
+	checkCmd := &cobra.Command{
+		Use:   "check",
+		Short: "Validate all files",
+		PersistentPreRun: func(cmd *cobra.Command, args []string) {
+			internal.InitializeFlags(cmd)
+		},
+		RunE: func(cmd *cobra.Command, args []string) error {
+			ctx := context.Background()
+
+			wd, err := os.Getwd()
+			if err != nil {
+				return fmt.Errorf("failed to get working directory: %w", err)
+			}
+
+			config, err := internal.ReadConfig(wd)
+			if err != nil {
+				return fmt.Errorf("failed to read config: %w", err)
+			}
+
+			migrationsDir, err := internal.GetMigrationsDir(wd)
+			if err != nil {
+				return fmt.Errorf("failed to get migrations directory: %w", err)
+			}
+
+			tmpDir, err := os.MkdirTemp("", "trek-")
+			if err != nil {
+				return fmt.Errorf("failed to create temporary directory: %w", err)
+			}
+			defer os.RemoveAll(tmpDir)
+
+			return checkAll(ctx, config, wd, tmpDir, migrationsDir)
+		},
+	}
+
+	return checkCmd
+}
+
+//nolint:cyclop
+func checkAll(
+	ctx context.Context,
+	config *internal.Config,
+	wd,
+	tmpDir,
+	migrationsDir string,
+) error {
+	postgres, conn, dsn, err := setupDatabase(ctx, tmpDir, "check", 5434)
+	defer func() {
+		if conn != nil {
+			_ = conn.Close(ctx)
+		}
+		if postgres != nil {
+			_ = postgres.Stop()
+		}
+	}()
+	if err != nil {
+		return fmt.Errorf("failed to setup database: %w", err)
+	}
+	dsn = fmt.Sprintf("%s?sslmode=disable", dsn)
+
+	for _, u := range config.DatabaseUsers {
+		var userExists bool
+		userExists, err = internal.CheckUserExists(ctx, conn, u)
+		if err != nil {
+			return fmt.Errorf("failed to check if user exists: %w", err)
+		}
+		if !userExists {
+			_, err = conn.Exec(ctx, fmt.Sprintf("CREATE ROLE %q WITH LOGIN PASSWORD 'postgres'", u))
+			if err != nil {
+				return fmt.Errorf("failed to create user: %w", err)
+			}
+		}
+	}
+
+	migrationFiles, err := internal.FindMigrations(migrationsDir, true)
+	if err != nil {
+		return fmt.Errorf("failed to find migrations: %w", err)
+	}
+
+	hookOptions := &internal.HookOptions{
+		Env: map[string]string{
+			"TREK_POSTGRES_HOST":     "localhost",
+			"TREK_POSTGRES_PORT":     "5434",
+			"TREK_POSTGRES_USER":     "postgres",
+			"TREK_POSTGRES_PASSWORD": "postgres",
+			"TREK_POSTGRES_DATABASE": "postgres",
+			"TREK_POSTGRES_SSLMODE":  "disable",
```

Comment:
Why are there hardcoded options?

Reply:
trek uses embedded databases for diffing and checking and those are just the default settings for the check database. they need to be hardcoded that way.